src/main/kotlin/com/netflix/dgs/plugin/hints/DgsInputArgumentInspector.kt

```diff
@@ -0,0 +1,237 @@
+                                inputArguments = "$inputArguments $parameter,"
+                                if (parameter != null) {
+                                    inputArgumentsList.add(parameter)
+                                }
+                            }
+                            inputArguments = inputArguments.removeSuffixIfPresent(",")
+
+                            val message = MyBundle.getMessage(
+                                "dgs.inspection.dgsinputargument.hint",
+                                inputArguments
+                            )
+                            holder.registerProblem(
+                                    node.parameterList.navigationElement,
+                                message,
+                                ProblemHighlightType.WEAK_WARNING,
+                                DgsInputArgumentQuickFix((node.toUElement() as UMethod), inputArgumentsList, message)
+                            )
+                        }
+                    }
+                }
+
+                println("DgsInputArgumentInspector took $time ms")
+                return super.visitMethod(node)
+            }
+        }, false)
+    }
+
+    private fun getHintForInputArgument(input: GraphQLInputValueDefinition, isJavaFile: Boolean) : String? {
+        return if (isJavaFile) {
+            getHintForInputArgumentInJava(input)
+        } else {
+            getHintForInputArgumentInKotlin(input)
+        }
+    }
+
+    private fun getHintForInputArgumentInJava(input: GraphQLInputValueDefinition) : String? {
+        val argName = (input.nameIdentifier as GraphQLIdentifierImpl).name
+        var inputArgumentHint = "@InputArgument "
+        if (input.type is GraphQLListType) {
+            val collectionType = getCollectionType(input.type, true)
+            if (! isSimpleType(collectionType)) {
+                inputArgumentHint = "$inputArgumentHint(collectionType=$collectionType.class) "
+            }
+        }
+        inputArgumentHint = inputArgumentHint + getType(input.type, true)  + " " + argName
+        return inputArgumentHint
+    }
+
+    private fun getHintForInputArgumentInKotlin(input: GraphQLInputValueDefinition) : String? {
+        val argName = (input.nameIdentifier as GraphQLIdentifierImpl).name
+        var inputArgumentHint = "@InputArgument "
+        if (input.type is GraphQLListType) {
+            val collectionType = getCollectionType(input.type, false)
+            if (! isSimpleType(collectionType)) {
+                inputArgumentHint = "$inputArgumentHint(collectionType=$collectionType) "
+            }
+        }
+        inputArgumentHint = inputArgumentHint + argName + ": "+ getType(input.type, false)  + " "
+        return inputArgumentHint
+    }
+
+    private fun isSimpleType(typeName: String?) : Boolean {
+        return typeName.equals("String") || typeName.equals("Integer") || typeName.equals("Int") || typeName.equals("Boolean") || typeName.equals("Double")
```

Comment:
won’t this get an NPE when `typeName` is null? I’m surprised the compiler allows this even

Reply:
Fixed all the methods to `String` instead or `String?` where applicable. This avoid the check.